src/main/java/org/openrewrite/staticanalysis/CombineSemanticallyEqualCatchBlocks.java

```diff
@@ -1692,8 +1692,7 @@ public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations m
                             nullMissMatch(multiVariable.getTypeExpression(), compareTo.getTypeExpression()) ||
                             multiVariable.getVariables().size() != compareTo.getVariables().size() ||
                             multiVariable.getLeadingAnnotations().size() != compareTo.getLeadingAnnotations().size() ||
-                            doesNotContainSameComments(multiVariable.getPrefix(), compareTo.getPrefix()) ||
-                            doesNotContainSameComments(multiVariable.getVarargs(), compareTo.getVarargs())) {
+                            doesNotContainSameComments(multiVariable.getPrefix(), compareTo.getPrefix())) {
```

Comment:
Reading the change set and the recipe I'm wondering what made you make this change here as opposed to a nullsafe check in the `doesNotContainSameComments`. Any context towards that? At first glance it seems we now convert fewer cases even if they could be equal; was that intended or a misreading on my part?

Reply:
Hi, thanks for the review.
I have now reverted the change and made a nullsafe check in `doesNotContainSameComments`.

To be honest, I thought that the check for varargs comments is unnecessary here as we are comparing catch blocks, and I didn't think about features like anonymous classes (or similar constructions) where varargs declarations could occur.
That said, I am still not sure if there is a case where `multiVariable.getVarargs()` is not null, as the `CommentVisitor` is only used when two catch blocks are semantically equal as defined by `SemanticallyEqual.areEqual(...)` and for two anonymous class definitions, this seems to never be the case.

For example, the following two catch blocks could be combined:
``` JAVA
class Test {
    void method() {
        try {
        } catch (A ex) {
            String s = "foo";
            
            new Object() {
              public void test(String... test) {
              }
            };
        } catch (B ex) {
            String s = "foo";
            
            new Object() {
              public void test(String... test) {
              }
            };
        }
    }
}
```

However, this is currently not done.